src/core/config/type-cast/type-casters/number.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * @prettier
+ */
+const numberTypeCaster = (value) => {
+  const parsedValue = parseInt(value)
+  return isNaN(parsedValue) ? value : parsedValue
```

Comment:
Let's use `Number.isNaN` to avoid implicit conversions that global `isNaN` is performing.

```suggestion
  return Number.isNaN(parsedValue) ? value : parsedValue
```